diff --git a/gradle.properties b/gradle.properties
index 68194ddc9..6bd97d8d9 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -11,7 +11,7 @@ name = logisim-evolution
 # * For non-stable builds use anything that reflects build state (i.e. "dev", "beta1", "rc1").
 #
 # NOTE: if you use suffix, you MUST separate it from the version number with a hypen.
-version = 3.9.0
+version = 3.9.0-jonichoBasys3Edition
 
 # Project website URL (the shorter the better)
 url = https://github.com/logisim-evolution/
diff --git a/src/main/java/com/cburch/logisim/gui/start/AboutCredits.java b/src/main/java/com/cburch/logisim/gui/start/AboutCredits.java
index 6d6e7d35a..9f3b48cd0 100644
--- a/src/main/java/com/cburch/logisim/gui/start/AboutCredits.java
+++ b/src/main/java/com/cburch/logisim/gui/start/AboutCredits.java
@@ -54,6 +54,8 @@ class AboutCredits extends JComponent {
         .title(BuildInfo.displayName)
         .h2(String.format("Copyright \u00A92001-%s %s developers", BuildInfo.year, BuildInfo.name))
         .url(BuildInfo.url)
+        .h2("Modified Basys3 Version by jonicho")
+        .url("https://github.com/jonicho/logisim-evolution-basys3")
         .space()
         .h1(S.get("creditsDevelopedBy"))
         .text("David H. Hutchens")
